### PR TITLE
Issue #1796 Add a config to skip all preflight checks

### LINKED ISSUE #1796: Add single toggle for preflight checks
Have a single toggle for preflight checks to disable with single config set

### PR DESCRIPTION
**Pull request by anjannath**
Fixes #1796

### CODE REVIEW

File: cmd/crc/cmd/start.go [hunk 1]
```diff
@@ -63,8 +63,11 @@ func runStart(arguments []string) (*machine.StartResult, error) {
 
 	checkIfNewVersionAvailable(config.Get(cmdConfig.DisableUpdateCheck).AsBool())
 
-	if err := preflight.StartPreflightChecks(config); err != nil {
-		return nil, err
+	// If skip-all-preflight-checks is true then don't run the prefligh checks
+	if !config.Get(cmdConfig.SkipAllPreflightChecks).AsBool() {
```
Comment (gbraad) [inline cmd/crc/cmd/start.go:67]: add log that all preflights were explicitly skipped

**Review by gbraad — CHANGES_REQUESTED** [review]
needs a log and/or feedback to the CLI

File: cmd/crc/cmd/start.go (hunk 1 above)
Comment (anjannath) [reply cmd/crc/cmd/start.go:67]: Added info log as..  `All preflight checks are skipped`

File: cmd/crc/cmd/start.go [hunk 2]
```diff
@@ -63,8 +63,13 @@ func runStart(arguments []string) (*machine.StartResult, error) {
...
+		if err := preflight.StartPreflightChecks(config); err != nil {
+			return nil, err
+		}
+	} else {
+		logging.Info("All preflight checks are skipped")
```
Comment (gbraad) [inline cmd/crc/cmd/start.go:72]: can this be inverted ?

```
if config.Get(cmdConfig.SkipAllPreflightChecks).AsBool() {
    logging.Info("All preflight checks are skipped")
} else {
    ...
}
```

Why the inversion of the value? Having the output clearly visible explains the purpose of this.

**Review by gbraad — CHANGES_REQUESTED** [review]
invert if

File: cmd/crc/cmd/start.go (hunk 2 above)
Comment (anjannath) [reply cmd/crc/cmd/start.go:67]: Changed!

**Comment by guillaumerose** [comment]
You also need to do that for the daemon API. 

Duplicated code sadely. We need to find a way to avoid that. (not here if you want to avoid a long debate :))

**Comment by anjannath** [comment]
> You also need to do that for the daemon API.
> 
> Duplicated code sadely. We need to find a way to avoid that. (not here if you want to avoid a long debate :))

Thanks so much,  forgot about the daemon :( for now added the same function in the daemon, we do need to avoid duplicating this..

File: cmd/crc/cmd/start.go [hunk 3]
```diff
@@ -83,6 +83,14 @@ func runStart(arguments []string) (*machine.StartResult, error) {
...
 }
 
+func runOrSkipPreflightChecks(skip bool) error {
+	if skip {
+		logging.Info("All preflight checks are skipped")
+		return nil
+	}
+	return preflight.StartPreflightChecks(config)
+}
+
```
Comment (praveenkumar) [inline cmd/crc/cmd/start.go:93]: This function is duplication for start and handler, better to have it part of preflight package itself and make `StartPreflightChecks` less visible. you can take `skip and config` as parameter for this one.

Comment (anjannath) [reply cmd/crc/cmd/start.go]: Changed!

**Review by guillaumerose — APPROVED** [review]

**Comment by guillaumerose** [comment]
Rebased. Let's wait for the CI now.

**Pull request closed without merging**